### ISSUE TITLE
`(require (combine-in))` prints `#t`

### ISSUE THREAD

**Issue by zyrolasting** [racket/racket#4048]
**What version of Racket are you using?**
8.2 [cs]

**What program did you run?**
`racket -e '(require (combine-in))'`

**What should have happened?**
No output, just like any non-terminal case of `combine-in`.

**Please include any other relevant details**
I'm working on a require transformer that acts as a front end for a program search. The result may correctly expand to at least zero nested require specifications, which I represent as `` `(combine-in . ,results) ``.

`(require (combine-in))` in particular expands to `(#%require (just-meta 0))`, which also prints `#t`. Given the macro stepper output and the fact I cannot nest the `#%require` in another expression, I suspect the `#t` is printed during expansion.

**Comment by samth** [comment]
It doesn't seem to be during expansion:
```
[samth@huor:~/work/rcon2020 (master) plt] r
Welcome to Racket v8.3.0.5 [cs].
> (expand '(#%require (just-meta 0)))
#<syntax (#%require (just-meta 0))>
> (eval
    '(#%require (just-meta 0)))
#t
```

**Comment by zyrolasting** [comment]
It could be, but I wasn't sure how to verify that without wrapping the `#%require` invocation.

**Comment by samth** [comment]
I don't think this is workaround-able. I expect there's just a print somewhere.

**Comment by mflatt** [comment]
Fixed in 5f14872fe65846c112c0eac36719748ae296d6a2
